### PR TITLE
Mention dialog modal

### PR DESCRIPTION
**Pull request by Jovialiste82**
- [x]  Implement logic
- [x] Style modal
- [x] Persist scrolling position

### CODE REVIEW

**Comment by Jovialiste82** [comment]
- [ ]  Implement logic
- [ ] Style modal

**Comment by Jovialiste82** [comment]
Hey guys, so the issue here is still that whenever the user closes the dialog, he is taken back at the top of the HomePage, and not at the scrolling position of the Mention he clicked on to open the Dialog. I tried to use two packages named "react-scroll-restoration" and "react-router-scroll-memory" but it did not work. What I ended up trying is saving in Local Storage the scrollY position and the Mentions fetched so far so that when the browser detects that we are coming from the Dialog page, it will use the data inside Local Storage instead of fetching again the data and use the scrollY stored in LS to bring us to the Mention, or at least the section where the mention is. Unfortunately the "window.scrollTo" does not work the way I want and the user is taken back to the top. Even weirder, if at that moment, the user refreshes the page, then the browser scrolls back down to the Mention.

**Comment by Jovialiste82** [comment]
Finally ! I think it's working guys ! Can you  test it on your end ? Thx again @nalanart ! I was really hitting a roadblock here. While waiting for your confirmation that it works too for you, I'll start taking care of the styling and cleaning up the code.

**Comment by Jovialiste82** [comment]
BTW, interesting to note this : for some reason I do not get yet, if the mention URL is just /:id it does not work, so I had to switch back to /something/:id

File: client/src/layout/Scroller.js [hunk 1]
```diff
@@ -15,21 +15,6 @@ const Scroller = ({
   mentionDatas,
   setMentionDatas,
```
Comment (nalanart) [inline client/src/layout/Scroller.js]: Unused props and imports, make sure to take them out!

File: client/src/pages/MentionPage.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      .then((data) => {
+        setMention(data);
+        setOpenDialog(true);
+      })
+      .catch((err) => alert("Something went wrong"));
+  }, []);
+
+  const handleclick = () => {
+    setOpenDialog(false);
+  };
```
Comment (nalanart) [inline client/src/pages/MentionPage.js]: for MentionPage.js, should the url change when you close the modal?

**Review by nalanart — CHANGES_REQUESTED** [review]
nice job, seems to be working well! Just a small thing about the MentionPage, check my comment. Also, there are some unused imports in the files that can be cleaned up!

**Review by GuohaoOuyang — APPROVED** [review]
Looks good to me!

File: client/src/pages/MentionPage.js (hunk 2 above)
Comment (Jovialiste82) [reply client/src/pages/MentionPage.js]: Well spotted ! I took care of it

File: client/src/layout/Scroller.js (hunk 1 above)
Comment (Jovialiste82) [reply client/src/layout/Scroller.js]: Took care of it. Thx !

**Pull request merged**